token/transfer-hook-interface/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+//! Crate defining an interface for performing a hook on transfer, where the
+//! token program calls into a separate program with additional accounts after
+//! all other logic, to be sure that a transfer has accomplished all required
+//! preconditions.
+
+#![allow(clippy::integer_arithmetic)]
+#![deny(missing_docs)]
+#![cfg_attr(not(test), forbid(unsafe_code))]
+
+pub mod error;
+pub mod inline_spl_token;
+pub mod instruction;
+pub mod invoke;
+#[cfg(feature = "offchain-client")]
+pub mod offchain;
+pub mod processor;
+
+#[cfg(not(feature = "no-entrypoint"))]
+mod entrypoint;
+
+// Export current sdk types for downstream users building with a different sdk version
+pub use solana_program;
+use solana_program::pubkey::Pubkey;
+
+/// Namespace for all programs implementing transfer-hook
+pub const NAMESPACE: &str = "spl-transfer-hook-interface";
+
+/// Size for discriminator in account and instruction data
+pub const DISCRIMINATOR_LENGTH: usize = 8;
```

Comment:
Seeing this makes me feel uncomfortable with how TLV Discriminators work, since it seems like a footgun to have to always define the size of discriminators. But for now, this is sick!